File: core/src/main/java/mifos/org/mobilewallet/core/data/fineract/entity/mapper/UserEntityMapper.java
package mifos.org.mobilewallet.core.data.fineract.entity.mapper;

import javax.inject.Inject;
import javax.inject.Singleton;

import mifos.org.mobilewallet.core.data.fineract.entity.UserEntity;
import mifos.org.mobilewallet.core.domain.model.User;

/**
 * Created by naman on 17/6/17.
 */

@Singleton
public class UserEntityMapper {

    @Inject
    public UserEntityMapper() {}

    public User transform(UserEntity userEntity) {
        User user = new User();

        if (userEntity != null) {
            user.setUserId(userEntity.getUserId());
            user.setUserName(userEntity.getUserName());
            user.setAuthenticationKey(userEntity.getBase64EncodedAuthenticationKey());

        }
        return user;
    }
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/UserEntityMapper.kt
package org.mifos.mobilewallet.core.data.fineract.entity.mapper

import org.mifos.mobilewallet.core.data.fineract.entity.UserEntity
import org.mifos.mobilewallet.core.domain.model.user.User
import javax.inject.Inject

/**
 * Created by naman on 17/6/17.
 */
class UserEntityMapper @Inject constructor() {

    fun transform(userEntity: UserEntity?): User {
        val user = User()
        if (userEntity != null) {
            user.userId = userEntity.userId!!
            user.userName = userEntity.userName
            user.authenticationKey = userEntity.base64EncodedAuthenticationKey
        }
        return user
    }
}